internal/gengapic/genrest.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) 2021  Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
```

Comment:
nit: I'm slightly uncomfortable with having one file named `gengapic.go` and this one named `genrest.go`: we're always generating GAPICs. as the package name implies, with different impls for each transport. Consider some parallel naming structure along the lines of `gengrpc.go`/`genrest.go` or `transport_grpc.go`/`transport_rest.go`

Reply:
If you're okay with it, I'd like to defer moving code into a new `gengrpc.go` file until the immediately subsequent review, just to keep the scope well contained. I agree, splitting it up is the correct and helpful thing to do.